buildSrc/src/main/java/org/elasticsearch/gradle/ElasticsearchJavaPlugin.java

```diff
@@ -269,7 +276,7 @@ public static void configureTestTasks(Project project) {
                 "-XX:+HeapDumpOnOutOfMemoryError"
             );
 
-            test.getJvmArgumentProviders().add(() -> List.of("-XX:HeapDumpPath=$heapdumpDir"));
+            test.getJvmArgumentProviders().add(new SimpleCommandLineArgumentProvider("-XX:HeapDumpPath=$heapdumpDir"));
```

Comment:
Oh, there is a bug here due to my previous PR, can you fix? heapdumpDir is a variable here that needs to be resolved, not a literal in the setting value.

Reply:
Good catch! Fixed.